packages/sdk-react/src/components/receivables/InvoiceDetails/CreateReceivable/sections/components/Billing/YourVatDetailsForm.tsx

```diff
@@ -0,0 +1,98 @@
+import { Controller, useFormContext } from 'react-hook-form';
+
+import { CountryInvoiceOption } from '@/components/receivables/InvoiceDetails/CreateReceivable/components/CountryInvoiceOption';
+import { CreateReceivablesFormProps } from '@/components/receivables/InvoiceDetails/CreateReceivable/validation';
+import { useMoniteContext } from '@/core/context/MoniteContext';
+import { useRootElements } from '@/core/context/RootElementsProvider';
+import { useMyEntity } from '@/core/queries';
+import { t } from '@lingui/macro';
+import { useLingui } from '@lingui/react';
+import {
+  FormControl,
+  InputLabel,
+  Select,
+  CircularProgress,
+  MenuItem,
+  FormHelperText,
+  TextField,
+  Collapse,
+  Box,
+} from '@mui/material';
+
+export const YourVatDetailsForm = ({ disabled }: { disabled: boolean }) => {
+  const { i18n } = useLingui();
+  const { control } = useFormContext<CreateReceivablesFormProps>();
+
+  const { root } = useRootElements();
+
+  const { api, monite } = useMoniteContext();
+  const { data: entityVatIds, isLoading: isEntityVatIdsLoading } =
```

Comment:
What do you think, if we also added error case adn wrapp it with our toaster/notification system?

Reply:
That may not be ideal for this section as we are displaying the VAT ids in the menu dropdown after fetching, a toast can easily be missed and the UI still wouldn't look great if this happens which is very unlikely, an alternative if needed in the future is an error state on the UI (which can come in a later iteration if needed as this is not a new component but just abstracted away).

Reply:
(ported code)